w3af/plugins/auth/detailed.py

```diff
@@ -58,7 +58,18 @@ def login(self):
 
         try:
             functor = getattr(self._uri_opener, self.method)
-            functor(self.auth_url, data)
+
+            while True:
```

Comment:
If the web application has a redirect loop, then w3af will be here for ever, which is really bad.